<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>GeneratePasswordDialog</class>
 <widget class="QDialog" name="GeneratePasswordDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>300</width>
    <height>258</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Generate Password</string>
  </property>
  <layout class="QGridLayout" name="gridLayout" columnstretch="1,0,0,1">
   <item row="0" column="0" colspan="4">
    <layout class="QFormLayout" name="formLayout">
     <property name="labelAlignment">
      <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
     </property>
     <property name="formAlignment">
      <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
     </property>
     <item row="0" column="0">
      <widget class="QLabel" name="label1">
       <property name="text">
        <string>Use digits (0, 1, 2, ...):</string>
       </property>
       <property name="buddy">
        <cstring>checkBoxDigits</cstring>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QCheckBox" name="checkBoxDigits">
       <property name="checked">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QLabel" name="label2">
       <property name="text">
        <string>Use minus (-):</string>
       </property>
       <property name="buddy">
        <cstring>checkBoxMinus</cstring>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <widget class="QCheckBox" name="checkBoxMinus">
       <property name="checked">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item row="2" column="0">
      <widget class="QLabel" name="label3">
       <property name="text">
        <string>Use underline (_):</string>
       </property>
       <property name="buddy">
        <cstring>checkBoxUnderline</cstring>
       </property>
      </widget>
     </item>
     <item row="2" column="1">
      <widget class="QCheckBox" name="checkBoxUnderline">
       <property name="checked">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item row="3" column="0">
      <widget class="QLabel" name="label4">
       <property name="text">
        <string>Use space ( ):</string>
       </property>
       <property name="buddy">
        <cstring>checkBoxSpace</cstring>
       </property>
      </widget>
     </item>
     <item row="3" column="1">
      <widget class="QCheckBox" name="checkBoxSpace"/>
     </item>
     <item row="4" column="0">
      <widget class="QLabel" name="label5">
       <property name="text">
        <string>Use special symbols (!, $, %, &amp;&amp;, ...):</string>
       </property>
       <property name="buddy">
        <cstring>checkBoxSpecialSymbols</cstring>
       </property>
      </widget>
     </item>
     <item row="4" column="1">
      <widget class="QCheckBox" name="checkBoxSpecialSymbols">
       <property name="checked">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item row="5" column="0">
      <widget class="QLabel" name="label6">
       <property name="text">
        <string>Use brackets ([, ], {, }, (, ), &lt;, &gt;):</string>
       </property>
       <property name="buddy">
        <cstring>checkBoxBrackets</cstring>
       </property>
      </widget>
     </item>
     <item row="5" column="1">
      <widget class="QCheckBox" name="checkBoxBrackets">
       <property name="checked">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item row="6" column="0">
      <widget class="QLabel" name="label7">
       <property name="text">
        <string>Password length:</string>
       </property>
       <property name="buddy">
        <cstring>spinBoxPasswordLength</cstring>
       </property>
      </widget>
     </item>
     <item row="6" column="1">
      <widget class="QSpinBox" name="spinBoxPasswordLength">
       <property name="alignment">
        <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
       </property>
       <property name="suffix">
        <string> symbols</string>
       </property>
       <property name="minimum">
        <number>5</number>
       </property>
       <property name="maximum">
        <number>100</number>
       </property>
       <property name="value">
        <number>25</number>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item row="1" column="0" colspan="4">
    <widget class="QPushButton" name="buttonGenerate">
     <property name="text">
      <string>Generate</string>
     </property>
    </widget>
   </item>
   <item row="2" column="0" colspan="4">
    <widget class="QLineEdit" name="lineEditPassword">
     <property name="placeholderText">
      <string>Password</string>
     </property>
     <property name="clearButtonEnabled">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item row="3" column="0" colspan="4">
    <widget class="QProgressBar" name="progressBar">
     <property name="maximumSize">
      <size>
       <width>16777215</width>
       <height>10</height>
      </size>
     </property>
     <property name="maximum">
      <number>4</number>
     </property>
     <property name="value">
      <number>0</number>
     </property>
     <property name="textVisible">
      <bool>false</bool>
     </property>
    </widget>
   </item>
   <item row="4" column="0">
    <spacer name="spacer1">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>0</width>
       <height>0</height>
      </size>
     </property>
    </spacer>
   </item>
   <item row="4" column="1">
    <widget class="QPushButton" name="buttonOk">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="text">
      <string>OK</string>
     </property>
     <property name="default">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item row="4" column="2">
    <widget class="QPushButton" name="buttonCancel">
     <property name="text">
      <string>Cancel</string>
     </property>
    </widget>
   </item>
   <item row="4" column="3">
    <spacer name="spacer2">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>0</width>
       <height>0</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
 </widget>
 <tabstops>
  <tabstop>buttonGenerate</tabstop>
  <tabstop>lineEditPassword</tabstop>
  <tabstop>buttonOk</tabstop>
  <tabstop>buttonCancel</tabstop>
  <tabstop>checkBoxDigits</tabstop>
  <tabstop>checkBoxMinus</tabstop>
  <tabstop>checkBoxUnderline</tabstop>
  <tabstop>checkBoxSpace</tabstop>
  <tabstop>checkBoxSpecialSymbols</tabstop>
  <tabstop>checkBoxBrackets</tabstop>
  <tabstop>spinBoxPasswordLength</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>buttonOk</sender>
   <signal>clicked()</signal>
   <receiver>GeneratePasswordDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>114</x>
     <y>223</y>
    </hint>
    <hint type="destinationlabel">
     <x>58</x>
     <y>224</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonCancel</sender>
   <signal>clicked()</signal>
   <receiver>GeneratePasswordDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>243</x>
     <y>226</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>227</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
